### PR TITLE
KOGITO-6413 Review Quarkus runtime extensions Maven setup according t…

### PR DESCRIPTION
**Pull request by cristianonicolai**
…o guide

Assembly:
- https://github.com/kiegroup/optaplanner/pull/1731
- https://github.com/kiegroup/kogito-runtimes/pull/1788

### CODE REVIEW

**Review by evacchi — APPROVED** [review]

File: quarkus/extensions/kogito-quarkus-decisions-extension/kogito-quarkus-decisions/pom.xml [hunk 1]
```diff
@@ -64,30 +63,33 @@
...
                     <execution>
+                        <phase>compile</phase>
                         <goals>
                             <goal>extension-descriptor</goal>
                         </goals>
-                        <phase>compile</phase>
                         <configuration>
-                            <deployment>${project.groupId}:${project.artifactId}-deployment:${project.version}</deployment>
-                            <capabilities>
-                                <provides>kogito-decisions</provides>
```
Comment (ricardozanini) [inline quarkus/extensions/kogito-quarkus-decisions-extension/kogito-quarkus-decisions/pom.xml:85]: We can't remove the capabilities.

File: quarkus/extensions/kogito-quarkus-predictions-extension/kogito-quarkus-predictions/pom.xml [hunk 2]
```diff
@@ -35,30 +35,33 @@
...
                     <execution>
+                        <phase>compile</phase>
                         <goals>
                             <goal>extension-descriptor</goal>
                         </goals>
-                        <phase>compile</phase>
                         <configuration>
-                            <deployment>${project.groupId}:${project.artifactId}-deployment:${project.version}</deployment>
-                            <capabilities>
-                                <provides>kogito-predictions</provides>
```
Comment (ricardozanini) [inline quarkus/extensions/kogito-quarkus-predictions-extension/kogito-quarkus-predictions/pom.xml:56]: Same here. But we need to review this name. Should be namespaced like `org.kie.kogito.engine.*`

File: quarkus/extensions/kogito-quarkus-processes-extension/kogito-quarkus-processes/pom.xml [hunk 3]
```diff
@@ -41,30 +41,33 @@
...
           <execution>
+            <phase>compile</phase>
             <goals>
               <goal>extension-descriptor</goal>
             </goals>
-            <phase>compile</phase>
             <configuration>
-              <deployment>${project.groupId}:${project.artifactId}-deployment:${project.version}</deployment>
-              <capabilities>
-                <provides>kogito-processes</provides>
```
Comment (ricardozanini) [inline quarkus/extensions/kogito-quarkus-processes-extension/kogito-quarkus-processes/pom.xml:62]: Same here. `org.kie.kogito.engine.processes`

File: quarkus/extensions/kogito-quarkus-rules-extension/kogito-quarkus-rules/pom.xml [hunk 4]
```diff
@@ -37,30 +37,33 @@
...
                     <execution>
+                        <phase>compile</phase>
                         <goals>
                             <goal>extension-descriptor</goal>
                         </goals>
-                        <phase>compile</phase>
                         <configuration>
-                            <deployment>${project.groupId}:${project.artifactId}-deployment:${project.version}</deployment>
-                            <capabilities>
-                                <provides>kogito-rules</provides>
```
Comment (ricardozanini) [inline quarkus/extensions/kogito-quarkus-rules-extension/kogito-quarkus-rules/pom.xml:58]: `org.kie.kogito.engine.rules`

File: quarkus/extensions/kogito-quarkus-serverless-workflow-extension/kogito-quarkus-serverless-workflow/pom.xml [hunk 5]
```diff
@@ -74,36 +74,38 @@
...
           <execution>
+            <phase>compile</phase>
             <goals>
               <goal>extension-descriptor</goal>
             </goals>
-            <phase>compile</phase>
             <configuration>
-              <deployment>${project.groupId}:${project.artifactId}-deployment:${project.version}</deployment>
-              <capabilities>
-                <provides>kogito-serverless-workflow</provides>
```
Comment (ricardozanini) [inline quarkus/extensions/kogito-quarkus-serverless-workflow-extension/kogito-quarkus-serverless-workflow/pom.xml:101]: `org.kie.kogito.engine.serverless.workflow`. The base processor we are using also needs to be updated.

File: quarkus/extensions/kogito-quarkus-decisions-extension/kogito-quarkus-decisions/pom.xml (hunk 1 above)
Comment (cristianonicolai) [reply quarkus/extensions/kogito-quarkus-decisions-extension/kogito-quarkus-decisions/pom.xml:85]: @ricardozanini these are defined in the in the code no? https://github.com/kiegroup/kogito-runtimes/blob/main/quarkus/extensions/kogito-quarkus-decisions-extension/kogito-quarkus-decisions-deployment/src/main/java/org/kie/kogito/quarkus/decisions/deployment/DecisionsAssetsProcessor.java#L35

Comment (cristianonicolai) [reply quarkus/extensions/kogito-quarkus-decisions-extension/kogito-quarkus-decisions/pom.xml:85]: oh ok, my bad.

**Review by ricardozanini — COMMENTED** [review]
Now I think we need to update these classes: https://github.com/kiegroup/kogito-runtimes/tree/main/quarkus/addons/common/deployment/src/main/java/org/kie/kogito/quarkus/addons/common/deployment

File: quarkus/extensions/kogito-quarkus-predictions-extension/kogito-quarkus-predictions/pom.xml (hunk 2 above)
Comment (cristianonicolai) [reply quarkus/extensions/kogito-quarkus-predictions-extension/kogito-quarkus-predictions/pom.xml:56]: @ricardozanini do you think we really need to append _.engine_ ? Was thinking we could drop that part:
_org.kie.kogito.processes_ and so on seems fine to me.

**Comment by cristianonicolai** [comment]
Jenkins rerun kogito-examples tests

**Review by ricardozanini — APPROVED** [review]
I thought you would remove the `engine` prefix. :D

**Comment by cristianonicolai** [comment]
jenkins please retest this

**Comment by cristianonicolai** [comment]
Jenkins rerun kogito-runtimes tests

**Comment by cristianonicolai** [comment]
Jenkins rerun kogito-runtimes tests

**Comment by manstis** [comment]
Does this PR affect how addons should be configured too? 

Trusty has an active [addon](https://github.com/kiegroup/kogito-runtimes/tree/main/quarkus/addons/tracing-decision) (i.e. it has ongoing development). Should we really be looking at moving it into the extensions?

**Comment by ricardozanini** [comment]
> 

@manstis your add-on is already configured to work with the [correct extension](https://github.com/kiegroup/kogito-runtimes/blob/main/quarkus/addons/tracing-decision/deployment/src/main/java/org/kie/kogito/tracing/decision/quarkus/deployment/KogitoAddOnTracingDecisionProcessor.java#L29). Extensions in the Kogito context are wrappers for our code-gen modules. Add-ons extend the capability of a given engine.

I don't think you guys need to move to an extension.

**Comment by cristianonicolai** [comment]
Jenkins rerun kogito-runtimes tests

**Comment by cristianonicolai** [comment]
Jenkins rerun kogito-runtimes tests

**Comment by cristianonicolai** [comment]
Jenkins rerun kogito-apps tests

**Pull request merged**